File: src/0801-0900/875 - Koko Eating Bananas/koko_eating_bananas.rs
impl Solution {
    pub fn min_eating_speed(piles: Vec<i32>, h: i32) -> i32 {
        let mut min_spd = 1;
        let mut max_spd = piles.iter().max().unwrap().clone();

        while min_spd < max_spd {
            let speed = (min_spd + max_spd) / 2;
            let hours = piles
                .iter()
                .fold(0, |subtotal, &pile| subtotal + ((pile + speed - 1) / speed));

            if hours > h {
                min_spd = speed + 1;
            } else {
                max_spd = speed;
            }
        }

        min_spd
    }
}

File: src/0801-0900/875 - Koko Eating Bananas/koko_eating_bananas.py
import math


class Solution:
    def minEatingSpeed(self, p: list[int], h: int) -> int:
        left, right = 1, max(p)

        while left <= right:
            m = (left+right)//2
            # t = sum((i+m-1)//m for i in p)
            # t = sum(-(-i//m) for i in p)
            t = sum(math.ceil(i/m) for i in p)
            if t > h:
                left = m+1
            else:
                right = m-1
        return left
